elastic/kibana
# [AO] Move filter query up and add auto-complete functionality

Closes #162767

## Summary

This PR moves the filter query up, based on the [design](https://www.figma.com/file/WtGj2hHUPVGTpQjSs1TtTe/Rule-creation-experience?type=design&node-id=459-386412&mode=design&t=jClkb4wrePTrVMwS-0).

<img src="https://github.com/elastic/kibana/assets/12370520/18c9bf5b-744e-44a0-bd40-872794d849f3" width=400 >

## 🧪 How to test
- Create a threshold rule without a filter
- Check the rule saved object and make sure we don't have a filter there
- Edit the rule and add a filter and check the auto-complete works fine
- Make sure the filter is applied to the preview
- Check the rule saved object and make sure the filter is saved there correctly
